.github/workflows/build_images_uv.yaml

```diff
@@ -1,26 +1,33 @@
 name: Docker image CI
 
 on:
-  pull_request:
-    paths:
-      - "lumigator/**"
-      - ".github/**"
-    # synchronized is when you push new commits
-    types: ["opened", "synchronize"]
   push:
     branches:
-      - main
+      - "main"
+  pull_request:
+    branches:
+      - "**"
   # required to enable manual triggers on the GH web ui
   workflow_dispatch:
 
 jobs:
-  build:
+  build-image:
     name: Build Docker image
     runs-on: ubuntu-latest
-
     steps:
-      - name: Checkout
+      - name: Checkout code
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check modified paths
+        run: |
+          if ! git diff --name-only ${{ github.event.before }} ${{ github.event.pull_request.head.sha }} | grep -E '^lumigator/|^\.github/'; then
```

Comment:
Checks like these have a lot of fiddly edge cases and unfortunately have been nothing but trouble in past iterations. Why not use the built in file filtering in the GH syntax?

Reply:
Thanks. I just left if in draft as I was testing some things out but needed the GH actions to run.

I wanted to see if I could get it working and also be included as a required status check (which would require it to run each time). 

I’ll check out the file filtering, and was also going to look at https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif

Reply:
This makes the PR much neater: https://github.com/dorny/paths-filter (thanks @agpituk for pointing me to it).